src/progress-bar/styled-components.js

```diff
@@ -50,6 +52,40 @@ export const BarProgress = styled<StylePropsT>('div', props => {
   };
 });
 
+export const InfiniteBarProgress = styled<StylePropsT>('div', props => {
```

Comment:
Can this be combined with the `BarProgress` component above? It looks like everything is the same besides lines 67-85 which are additive. It can be done by including an optional `$infinite` boolean on `StylePropsT` where if it's set, add those additional lines